samples/cellular/modem_shell/README.rst

```diff
@@ -1052,45 +1052,85 @@ To program the certificates and connect to nRF Cloud, complete the following ste
-
-   west build -p -b *build_target* -- -DDTC_OVERLAY_FILE=ppp.overlay -DOVERLAY_CONFIG=overlay-ppp.conf
-
-|build_target|
+.. _modem_shell_extensions:
+
+Activating sample extensions
+============================
+
+You can enable the following optional extensions during the sample build process:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Extension
+     - Files and Kconfig options to use
+     - Build variables
+   * - nRF91 Series DK with nRF7002 EK Wi-Fi support
+     - :file:`overlay-nrf7002ek-wifi-scan-only.conf`
+     - * ``-DEXTRA_CONF_FILE=overlay-nrf7002ek-wifi-scan-only.conf``
+       * ``-DSHIELD=nrf7002ek``
+   * - ESP8266 Wi-Fi support
+     - * :file:`overlay-esp-wifi.conf`
+       * :file:`esp_8266_nrf9160ns.overlay`
+     - * ``-DEXTRA_CONF_FILE=overlay-esp-wifi.conf``
+       * ``-DEXTRA_DTC_OVERLAY_FILE=esp_8266_nrf9160ns.overlay``
+   * - PPP support (see also `Testing PPP support`_)
+     - * :file:`overlay-ppp.conf`
+       * :file:`ppp.overlay`
+     - * ``-DEXTRA_CONF_FILE=overlay-ppp.conf``
+       * ``-DEXTRA_DTC_OVERLAY_FILE=ppp.overlay``
+   * - FOTA support
+     - :file:`overlay-app_fota.conf`
+     - ``-DEXTRA_CONF_FILE=overlay-app_fota.conf``
+   * - nRF91 Series DK with full modem FOTA support
+     - * :file:`overlay-modem_fota_full.conf`
+       * :file:`<device>_ext_flash.overlay`
+     - * ``-DEXTRA_CONF_FILE=overlay-modem_fota_full.conf``
+       * ``-DEXTRA_DTC_OVERLAY_FILE=<device>_ext_flash.overlay``
+   * - LwM2M carrier library support
+     - :file:`overlay-carrier.conf`
+     - ``-DEXTRA_CONF_FILE=overlay-carrier.conf``
+   * - P-GPS support
+     - :file:`overlay-pgps.conf`
+     - ``-DEXTRA_CONF_FILE=overlay-pgps.conf``
+   * - Cloud over MQTT (remote control)
+     - :file:`overlay-cloud_mqtt.conf`
+     - ``-DEXTRA_CONF_FILE=overlay-cloud_mqtt.conf``
+   * - Cloud over CoAP
+     - :file:`overlay-cloud_coap.conf`
+     - ``-DEXTRA_CONF_FILE=overlay-cloud_coap.conf``
+   * - Location services in MoSh (instead of :ref:`lib_location`)
+     - * :file:`overlay-cloud_mqtt.conf`
+       * :kconfig:option:`CONFIG_LOCATION_SERVICE_EXTERNAL`
+     - * ``-DEXTRA_CONF_FILE=overlay-cloud_mqtt.conf``
+       * ``-DCONFIG_LOCATION_SERVICE_EXTERNAL=y``
+   * - Location services in MoSh, with P-GPS
+     - * :file:`overlay-cloud_mqtt.conf` and :file:`overlay-pgps.conf`
+       * :kconfig:option:`CONFIG_LOCATION_SERVICE_EXTERNAL`
+       * :kconfig:option:`CONFIG_NRF_CLOUD_PGPS_TRANSPORT_NONE`
+     - * ``-DEXTRA_CONF_FILE="overlay-cloud_mqtt.conf;overlay-pgps.conf"``
+       * ``-DCONFIG_LOCATION_SERVICE_EXTERNAL=y``
+       * ``-DCONFIG_NRF_CLOUD_PGPS_TRANSPORT_NONE=y``
+   * - SEGGER Real Time Transfer (RTT) support (instead of UART)
+     - :file:`overlay-rtt.conf`
+     - ``-DEXTRA_CONF_FILE=overlay-rtt.conf``
+
+.. |variable_feature| replace:: the FOTA support
+.. |makevar| replace:: EXTRA_CONF_FILE
+.. |cmake_file_name| replace:: overlay-app_fota.conf
+.. |board_name| replace:: nrf9160dk_nrf9160_ns
```

Comment:
I'd like to take an example for a newer chipset which would be nrf9161 at the moment.